MotionMark/tests/resources/main.js

```diff
@@ -376,16 +378,27 @@ AdaptiveController = Utilities.createSubclass(Controller,
         this._intervalFrameCount = 0;
         this._intervalTimestamp = timestamp;
     }
-});
+}
 
-RampController = Utilities.createSubclass(Controller,
-    function(benchmark, options)
-    {
-        this.targetFPS = options["frame-rate"];
+class RampController extends Controller {
+    // If the engine can handle the tier's complexity at the desired frame rate, test for a short
+    // period, then move on to the next tier
+    static tierFastTestLength = 250;
+    // If the engine is under stress, let the test run a little longer to let the measurement settle
+    static tierSlowTestLength = 750;
+    // Tier intervals must have this number of non-outlier frames in order to end.
+    static numberOfFramesRequiredInInterval = 9;
 
+    static rampWarmupLength = 200;
+    
+    constructor(benchmark, options)
+    {
+        const targetFPS = options["frame-rate"];
         // The tier warmup takes at most 5 seconds
-        options["sample-capacity"] = (options["test-interval"] / 1000 + 5) * this.targetFPS;
-        Controller.call(this, benchmark, options);
+        options["sample-capacity"] = (options["test-interval"] / 1000 + 5) * targetFPS;
```

Comment:
The caller should make this calculation before constructing the `RampController`. So we can call `super(...)` the first thing in this constructor.